src/app/detail/ui/ArchiveWrapper.tsx

```diff
@@ -0,0 +1,66 @@
+'use client'
+
+import { ArchiveBtn } from '@/components/buttons/ArchiveBtn'
+import { useEffect, useState } from 'react'
+import useSWR, { useSWRConfig } from 'swr'
+import { fetcher } from '@/utils/fetcher'
+import { useSelector } from 'react-redux'
+import { selectTaskBoard } from '@/redux/features/taskBoardSlice'
+import { Skeleton } from '@mui/material'
+import store from '@/redux/store'
+import { setTask } from '@/redux/features/taskDetailsSlice'
+
+export const ArchiveWrapper = ({ taskId }: { taskId: string }) => {
+  const { token } = useSelector(selectTaskBoard)
+  const { mutate } = useSWRConfig()
+  const cacheKey = `/api/tasks/${taskId}?token=${token}`
+  const { data, error } = useSWR(cacheKey, fetcher)
+  const [isArchived, setIsArchived] = useState<boolean | undefined>(undefined)
+
+  // Set the initial state when `data` becomes available
+  useEffect(() => {
+    if (data?.task) {
+      setIsArchived(data.task.isArchived)
+      store.dispatch(setTask(data.task))
+    }
+  }, [data])
+
+  const handleToggleArchive = async () => {
+    if (isArchived === undefined) return // Prevent toggling if state isn't initialized yet
+
+    const newIsArchived = !isArchived
+    const optimisticTask = { ...data.task, isArchived: newIsArchived }
+
+    setIsArchived(newIsArchived) // Optimistically update local state
+
+    try {
+      await mutate(
+        cacheKey,
+        async () => {
+          // Call the actual API to toggle isArchived
+          await fetch(`/api/tasks/${taskId}?token=${token}`, {
+            method: 'PATCH',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ isArchived: newIsArchived }),
+          })
+
+          // Re-fetch updated data
+          return await fetcher(cacheKey)
+        },
+        {
+          optimisticData: { task: optimisticTask }, // Update UI immediately
+          rollbackOnError: true, // Revert on error
+          revalidate: true, // Ensure data consistency
+        },
```

Comment:
How are we checking for realtime archive updates?

Reply:
Removed `revalidate:true` in the commit below